apps/web/lib/contexts/LanguageContext.tsx

```diff
@@ -0,0 +1,25 @@
+import { createContext, useEffect, useState } from "react";
+
+interface LanguageContextInterface {
+  currentLang: string;
+  setLang: (newLanguage: string) => void;
+}
+
+export const LanguageContext = createContext<LanguageContextInterface>(
+  undefined!
+);
+
+export const LanguageProvider = ({ children }: any) => {
+  const [lang, setLang] = useState("en");
+
```

Comment:
```suggestion
interface LanguageContextInterface {
  currentLang: string;
  setLang: Dispatch<SetStateAction<string>>;
}

export const LanguageContext = createContext<LanguageContextInterface>(
  currentLang: "",
  setLang: () => {},
);

export const LanguageProvider = ({ children }: React.ReactNode) => {
  const [lang, setLang] = useState("en");

```
I'm thinking of a fixed, more specific type for setLang, and also initialize the context with the right properties, not undefined!